vpr/src/route/connection_box_lookahead_map.cpp

```diff
@@ -537,6 +751,91 @@ float ConnectionBoxMapLookahead::get_expected_cost(
     }
 }
 
+static vtr::Rect<int> bounding_box_for_node(const t_rr_node& node) {
+    return vtr::Rect<int>(node.xlow(), node.ylow(),
+                          node.xhigh() + 1, node.yhigh() + 1);
+}
+
+static vtr::Point<int> point_for_node(const t_rr_node& node) {
+    int x = (node.xhigh() + node.xlow()) / 2;
+    int y = (node.yhigh() + node.ylow()) / 2;
+    return vtr::Point<int>(x, y);
+}
+
+// for each segment type, find the nearest nodes to an equally spaced grid of points
+// within the bounding box for that segment type
+static void find_inodes_for_segment_types(std::vector<SampleGrid>* inodes_for_segment) {
+    auto& device_ctx = g_vpr_ctx.device();
+    auto& rr_nodes = device_ctx.rr_nodes;
+    const int num_segments = inodes_for_segment->size();
+
+    // compute bounding boxes for each segment type
+    std::vector<vtr::Rect<int>> bounding_box_for_segment(num_segments, vtr::Rect<int>());
+    for (size_t i = 0; i < rr_nodes.size(); i++) {
+        auto& node = rr_nodes[i];
+        if (node.type() != CHANX && node.type() != CHANY) continue;
+        int seg_index = device_ctx.rr_indexed_data[node.cost_index()].seg_index;
+
+        VTR_ASSERT(seg_index != OPEN);
+        VTR_ASSERT(seg_index < num_segments);
+
+        bounding_box_for_segment[seg_index] |= bounding_box_for_node(node);
+    }
+
+    // select an inode near the center of the bounding box for each segment type
```

Comment:
This comment seems like it is on the wrong loop?

Reply:
Fixed.